### PR TITLE
Update content retrieval from AutoModActionExecution

### PR DESCRIPTION
**Pull request by NeloBlivion**
Update to fix `data["content"]` raising a KeyError for Automod Events when the bot [lacks the Message Content intent](https://discord.com/developers/docs/change-log#jun-21-2022).

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Review by Lulalaby — APPROVED** [review]

File: discord/raw_models.py [hunk 1]
```diff
@@ -459,9 +459,9 @@ def __init__(self, state: ConnectionState, data: AutoModActionExecution) -> None
         self.guild_id: int = int(data["guild_id"])
         self.guild: Optional[Guild] = state._get_guild(self.guild_id)
         self.user_id: int = int(data["user_id"])
-        self.content: str = data["content"]
+        self.content: str = data.get("content", None)
```
Comment (Dorukyum) [inline discord/raw_models.py:462]:
```suggestion
        self.content: Optional[str] = data.get("content", None)
```

File: discord/raw_models.py [hunk 2]
```diff
@@ -459,9 +459,9 @@ def __init__(self, state: ConnectionState, data: AutoModActionExecution) -> None
...
         self.matched_keyword: str = data["matched_keyword"]
-        self.matched_content: str = data["matched_content"]
+        self.matched_content: str = data.get("matched_content", None)
```
Comment (Dorukyum) [inline discord/raw_models.py:464]:
```suggestion
        self.matched_content: Optional[str] = data.get("matched_content", None)
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by Lulalaby — APPROVED** [review]

**Comment by Lulalaby** [comment]
Ah right

**Comment by Yuvaraja28** [comment]
Here's the Full Traceback
traceback

**Review by Dorukyum — APPROVED** [review]
Thanks

**Pull request merged**